logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,39 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
+require "logstash/settings"
 
 module LogStash
+
+  [
+            Setting::String.new("node.name", Socket.gethostname),
+            Setting::String.new("config.path", nil, false),
+            Setting::String.new("config.string", nil, false),
+           Setting::Boolean.new("config.test", false),
+           Setting::Boolean.new("config.reload.auto", false),
+           Setting::Numeric.new("config.reload.interval", 3),
+           Setting::Boolean.new("metric.collect", true) {|v| v == true }, # metric collection cannot be disabled
+            Setting::String.new("settings.dir", ::File.join(Environment::LOGSTASH_HOME, "config")),
```

Comment:
`path.settings`? this way it will be consistent with ES and we are surely going to have `path.data`. I think this will round it up nicely:

`path.data`
`path.settings`
`path.logs`